### ISSUE TITLE
COPY command

### ISSUE THREAD

**Issue by yonigibbs** [reactiverse/reactive-pg-client#153]
Hi @vietj.  Is it possible to run a [COPY](https://www.postgresql.org/docs/10/static/sql-copy.html) command through reactive-pg-client?  Specifically, to load a CSV of data in what I believe is the most efficient way to load a large number of records into the system.  I believe the CSV has to either be on the DB server itself (which we'd like to avoid if possible), so can we in some way use the STDIN argument of the COPY command to provide the CSV file over the wire to it?

Thanks!

**Comment by vietj** [comment]
@yonigibbs no it has not been implemented as far as I know, this is a part of the postgresl protocol I'm not familiar (yet) with - but it is planned to be implemented

**Comment by vietj** [comment]
copy seems indeed to be a stream oriented operation (in both ways), I believe it can be implemented fairly easily.

**Comment by yonigibbs** [comment]
Great, thanks. We'll keep an eye out for it being implemented in future.

**Comment by davidzzzzz** [comment]
@vietj did anyone volunteer for this? Otherwise, I think we'd like to take it on

**Comment by vietj** [comment]
@davidzzzzz it's all yours if you take it

**Comment by davidzzzzz** [comment]
Great, on it

**Comment by davidzzzzz** [comment]
@vietj Here's the direction I've taken with supporting `COPY FROM` operations. Please provide feedback before I add more documentation and polish. Thanks!

https://github.com/reactiverse/reactive-pg-client/pull/205

**Comment by bchapuis** [comment]
Hi @vietj and @davidzzzzz, I'm really interested in experimenting with `COPY FROM`. As #205 is closed what's the current status?

**Comment by Globegitter** [comment]
Yeah same here @vietj - what exactly happened with the PR that added the `COPY` command? IT seems it got swallowed by the 3.8 release. Any plans to bring it back?

**Comment by vietj** [comment]
@Globegitter I think it simply got lost, it should be rebased on latest 3.8 indeed

**Comment by davidzzzzz** [comment]
If there's interest @vietj, I'm happy to revive this

**Comment by vietj** [comment]
yes there is, sorry for letting this dormant

**Comment by Globegitter** [comment]
@davidzzzzz any chance this can be revived again?

**Comment by davidzzzzz** [comment]
@Globegitter sorry about that, my day job got in the way :) Hoping to get to this shortly

**Comment by Globegitter** [comment]
@davidzzzzz No worries, I know the feeling :)

**Comment by Ryaryu** [comment]
I know it is annoying to bump this just to ask, but you guys got the time to revive this?

**Comment by jlnme** [comment]
Hello everyone, I would also love this feature to be included. Any chance we get this?

**Comment by vietj** [comment]
I set help-wanted on this issue as we need a contributor to step up and implement it.

**Comment by psolomin** [comment]
Hello @vietj 

Are you aware if anyone is working on this at the moment? If no, I would like to assign myself to this.

**Comment by vietj** [comment]
nobody is currently contributing this, so you can definitely take it.

you should come on our discord to discuss the feature
